### PR TITLE
feat(site): add a provisioner warning to workspace builds

### LINKED ISSUE #15048: Improve UX when there is no corresponding provisioner for a workspace/template
Since more and more of our users are using external provisioners and [provisioner tags](https://coder.com/docs/admin/provisioners#provisioner-tags), especially with Organizations, we should explore ways to improve the UX when a provisioner is not running. 

### Create/duplicate template

Image

### Template Editor

https://github.com/user-attachments/assets/fa1f5c2c-e965-4009-af8b-ed8113a95069

> Suggestion: "No provisioners are running with the tag `provisioner=missing` in the `data-science` org. Learn more about provisioners in [the Coder documentation](https://coder.com/docs/admin/provisioners)"

## Create workspace

Image

> Suggestion: "No provisioners are running with the tag `provisioner=missing` in the `data-science` org.
> Learn more about provisioners in [the Coder documentation](https://coder.com/docs/admin/provisioners)"

## Desired state

- Some meaningful and actionable error. One suggestion:
    > No provisioners are running with the tag `provisioner=missing` in the `data-science` org
    > Learn more about provisioners in [the Coder documentation](https://coder.com/docs/admin/provisioners)"
- Perhaps a custom troubleshooting URL for users, similar to the [agent](https://registry.terraform.io/providers/coder/coder/latest/docs/resources/agent#troubleshooting_url) as the organization/template may have internal docs about provisioners work in their organization.

### PR DESCRIPTION
**Pull request by SasSwart**
This PR adds warnings about provisioner health to workspace build pages.
It closes https://github.com/coder/coder/issues/15048

image
image

### CODE REVIEW

File: site/src/pages/WorkspacePage/Workspace.stories.tsx [hunk 1]
```diff
@@ -95,6 +95,51 @@ export const PendingInQueue: Story = {
...
+};
+
+export const PendingWithUndefinedProvisioners: Story = {
+	args: {
+		...Running.args,
+		workspace: {
+			...Mocks.MockPendingWorkspace,
+			latest_build: {
+				...Mocks.MockPendingWorkspace.latest_build,
+				matched_provisioners: undefined,
```
Comment (mtojek) [inline site/src/pages/WorkspacePage/Workspace.stories.tsx:137]: I am curious: when is this a valid/expected state?

File: site/src/modules/provisioners/ProvisionerStatusAlert.tsx [hunk 2]
```diff
@@ -1,17 +1,19 @@
 import type { AlertColor } from "components/Alert/Alert";
 import type { FC } from "react";
-import { ProvisionerAlert } from "./ProvisionerAlert";
+import { AlertVariant, ProvisionerAlert } from "./ProvisionerAlert";
 
 interface ProvisionerStatusAlertProps {
 	matchingProvisioners: number | undefined;
 	availableProvisioners: number | undefined;
 	tags: Record<string, string>;
+	variant?: AlertVariant;
```
Comment (mtojek) [inline site/src/modules/provisioners/ProvisionerStatusAlert.tsx:9]: Should we mark `variant` non optional?

File: site/src/modules/provisioners/ProvisionerAlert.tsx [hunk 3]
```diff
@@ -4,30 +4,38 @@ import { AlertDetail } from "components/Alert/Alert";
 import { Stack } from "components/Stack/Stack";
 import { ProvisionerTag } from "modules/provisioners/ProvisionerTag";
 import type { FC } from "react";
+
+export enum AlertVariant {
+	Standalone = "Standalone",
+	InLogs = "InLogs",
+}
```
Comment (mtojek) [inline site/src/modules/provisioners/ProvisionerAlert.tsx:11]: You may want to comment on the differences between these two options and describe how they are rendered.

File: site/src/modules/provisioners/ProvisionerAlert.tsx [hunk 4]
```diff
@@ -4,30 +4,38 @@ import { AlertDetail } from "components/Alert/Alert";
...
 	severity,
 	tags,
+	variant = AlertVariant.Standalone,
 }) => {
 	return (
 		<Alert
 			severity={severity}
-			css={(theme) => {
-				return {
+			{...(variant === AlertVariant.InLogs && {
```
Comment (mtojek) [inline site/src/modules/provisioners/ProvisionerAlert.tsx:31]: nit (readability): I think you can extract it to a separate function or move up to a const.

File: site/src/modules/provisioners/ProvisionerAlert.stories.tsx [hunk 5]
```diff
@@ -21,6 +21,26 @@ export default meta;
 type Story = StoryObj<typeof ProvisionerAlert>;
 
 export const Info: Story = {};
+
+export const InfoStyledForLogs: Story = {
+	args: {
+		variant: AlertVariant.InLogs,
```
Comment (mtojek) [inline site/src/modules/provisioners/ProvisionerAlert.stories.tsx:27]: My main question is about naming - does "InLogs" reflect variant's behavior or location? Maybe we should consider a bit different name, something that is not referring to the place where is used but rather its function (no frame?)

File: site/src/pages/WorkspacePage/Workspace.stories.tsx (hunk 1 above)
Comment (SasSwart) [reply site/src/pages/WorkspacePage/Workspace.stories.tsx:137]: https://github.com/coder/coder/blob/32aec0b07be5596403f56d81f7cad8114ff17db1/site/src/testHelpers/entities.ts#L1202-L1206

File: site/src/modules/provisioners/ProvisionerStatusAlert.tsx (hunk 2 above)
Comment (SasSwart) [reply site/src/modules/provisioners/ProvisionerStatusAlert.tsx:9]: The component uses a reasonable default for the variant. So I think optional is fine. Is there a specific concern that could be alleviated by making it non-optional?

File: site/src/pages/WorkspacePage/Workspace.stories.tsx (hunk 1 above)
Comment (SasSwart) [reply site/src/pages/WorkspacePage/Workspace.stories.tsx:137]: The values would depend on what a user has configured, but this is one example of a healthy payload.

File: site/src/modules/provisioners/ProvisionerAlert.stories.tsx (hunk 5 above)
Comment (SasSwart) [reply site/src/modules/provisioners/ProvisionerAlert.stories.tsx]: updated

File: site/src/modules/provisioners/ProvisionerAlert.tsx (hunk 4 above)
Comment (SasSwart) [reply site/src/modules/provisioners/ProvisionerAlert.tsx]: done

File: site/src/modules/provisioners/ProvisionerAlert.stories.tsx [hunk 6]
```diff
@@ -21,6 +21,26 @@ export default meta;
 type Story = StoryObj<typeof ProvisionerAlert>;
 
 export const Info: Story = {};
+
+export const InfoStyledForLogs: Story = {
```
Comment (mtojek) [inline site/src/modules/provisioners/ProvisionerAlert.stories.tsx:25]: nit: you can rename this too: `InfoInline`

File: site/src/modules/provisioners/ProvisionerAlert.stories.tsx [hunk 7]
```diff
@@ -21,6 +21,26 @@ export default meta;
...
+	},
+};
+
+export const Warning: Story = {
+	args: {
+		severity: "warning",
+	},
+};
+
+export const WarningStyledForLogs: Story = {
```
Comment (mtojek) [inline site/src/modules/provisioners/ProvisionerAlert.stories.tsx:37]: same

**Review by mtojek — APPROVED** [review]
Left 2 tiny nitpicks.

LGTM  👍

**Review by mtojek — APPROVED** [review]
Ship it! Thanks for all changes.

**Comment by SasSwart** [comment]
Thanks @mtojek. The frontend works, but in PR env testing I noticed the backend doesn't populate the fields where it needs to. One more commit just to fix that and then we'll ship. Apologies for the premature review.

**Review by johnstcn — APPROVED** [review]

**Review by johnstcn — APPROVED** [review]
:shipit:

**Pull request merged**